### ISSUE TITLE
Source code request

### ISSUE THREAD

**Issue by jja2000** [err4o4/spotify-car-thing-reverse-engineering#9]
Hi! Has anyone tried asking for the source code for whatever form of Linux they're running from Spotify themselves?

Might be some good pointers in there for how the hardware is wired up internally.

**Comment by jja2000** [comment]
Had no success using the regular support. Made a thread on the developer forum: https://community.spotify.com/t5/Spotify-for-Developers/Open-Source-Software-on-the-Car-Thing/m-p/5443263

**Comment by rickdeck** [comment]
Apparently the GPL source was released in 2021 here: https://github.com/spsgsb

**Comment by jja2000** [comment]
> Apparently the GPL source was released in 2021 here: https://github.com/spsgsb

It's honestly interesting that I can't find anything linking to this account except for this repo. I'll look through it though, thanks!

**Comment by rickdeck** [comment]
Yeah, it's weird. I assume the device was developed by an ODM who just created the git to comply with GPL obligations and be done with it.

If it would have been in the hands of Spotify, I'd assume they would have simply provided the source as a zip somewhere on their page (like many OEM do)

**Comment by everettperiman** [comment]
Is anyone familiar on if it is possible to simulate a swupdate server? The .swu that was found includes shell scripts for pre/post install. Depending on what Error correction the SWU process has it might be possible to inject our own code into the scripts and get shell access.
